### PR TITLE
[Doc Rework] Include expressions into API.md file

### PR DESCRIPTION
**Pull request by mapedraza**
This PR is part of a bigger series https://github.com/telefonicaid/iotagent-node-lib/pull/1332

### CODE REVIEW

**Comment by mapedraza** [comment]
Apart from the modifications required after solving this thread:
https://github.com/telefonicaid/iotagent-node-lib/pull/1357#discussion_r1184792968

This PR is ready to review

File: doc/api.md [hunk 1]
```diff
@@ -452,20 +460,408 @@ Note that, even creating a group with `autoprovision=true` and `explicitAttrs=tr
...
+            "type": "Number"
+        }
+    ]
+}
+```
+
+When receiving the attributes `v`, `v1` and `v2` in a payload from a message received in the southbound:
+
+```json
+({
```
Comment (fgalan) [inline doc/api.md:800]: What `(` means here?

File: doc/api.md [hunk 2]
```diff
@@ -487,17 +883,17 @@ translation from geo:point to geo:json)
...
           "type":"geo:point",
-          "expression": "${@latitude}, ${@longitude}",
+          "expression": "latitude, longitude",
           "reverse": [
             {
               "object_id":"longitude",
-              "type": "Text",
-              "expression": "${trim(substr(@location, indexOf(@location, \",\") + 1, length(@location)))}"
+              "type": "Number",
+              "expression": "location | split(', ')[0] | parsefloat()"
```
Comment (fgalan) [inline doc/api.md:891]: Looking this kind of changes... should we understand that all the mentions to the old legacy expression legacy has been removed from documentation?

Comment (mapedraza) [reply doc/api.md:891]: For documents processed yes. This should be checked again in dedicated PR https://github.com/telefonicaid/iotagent-node-lib/pull/1339

File: doc/api.md [hunk 3]
```diff
@@ -452,20 +460,408 @@ Note that, even creating a group with `autoprovision=true` and `explicitAttrs=tr
...
+            "object_id": "v",
+            "name": "vol",
+            "expression" : "v*100",
+            "type": "Number"
+        }
+    ]
+}
+```
+
+When receiving the attributes `v`, `v1` and `v2` in a payload from a message received in the southbound:
```
Comment (fgalan) [inline doc/api.md:797]: In fact, what are received are *measure* not *attributes*. Attributes is the conversion of the measure in the southbound to the Context Broker context in the northbound.

Maybe better:

```suggestion
When receiving the measures `v`, `v1` and `v2` in a payload from a message received in the southbound:
```

File: doc/api.md [hunk 4]
```diff
@@ -452,20 +460,408 @@ Note that, even creating a group with `autoprovision=true` and `explicitAttrs=tr
...
+{
+    "name": "v1",
+    "type": "Number",
+    "value": 1
+},
+{
+    "name": "v2",
+    "type": "Number",
+    "value": 2
+}
```
Comment (fgalan) [inline doc/api.md:814]: As far as I understand no IOTA using this library uses this payload at southbound for measures. The closer one is IOTA JSON, that would be:

```suggestion
{
    "v": 0
},
{
    "v1": 1
},
{
    "v2": 2
}
```

However, not sure if we should even use a particular southbound payload in the example or describe the reception of measures in an transport-agnostic way...

File: doc/api.md (hunk 3 above)
Comment (mapedraza) [reply doc/api.md:797]: I Agree, better to use measure term

File: doc/api.md (hunk 4 above)
Comment (mapedraza) [reply doc/api.md]: done in https://github.com/telefonicaid/iotagent-node-lib/pull/1359/commits/0ed3d13313837524cc6f1f7995b7df57f520c9e0

**Review by fgalan — APPROVED** [review]
LGTM

File: doc/api.md (hunk 2 above)
Comment (fgalan) [reply doc/api.md:886]: NTC

**Pull request merged**